cfme/web_ui/__init__.py

```diff
@@ -3994,6 +3994,24 @@ def controller_name():
     return sel.execute_script('return ManageIQ.controller;')
 
 
+def match_location(controller=None, title=None, summary=None):
+    """Does exact match of passed data
+
+        Returns:
+        :py:class:`bool`
+    """
+    result = []
+    if controller:
+        result.append(controller_name() == controller)
+    if title:
+        result.append(browser_title() == title)
+    if summary:
+        result.append(summary_title() == summary or
+                      sel.is_displayed('//h3[text() == "{}"]'.format(summary)))
```

Comment:
Couple of rules when it comes to XPath and HTML:

- there is no `==`
- when matching text, always use normalize-space: `normalize-space(.)="foo"` or `normalize-space(text())="foo"`